### DISCUSSION TITLE
Scan Orientation vs Placing on wall, ceiling, horizontal

### DISCUSSION THREAD

**Discussion by sugarfreejaz** [google/model-viewer#4583] [Q&A]
I have scanned some objects using Scaniverse.  Scanned the objects on horizontal surface.  I would like the object to be placed on walls and ceilings, which I have accomplished.  However, I need to object to be placed on the wall offset 90 degrees.  The bottom of the object needs to be placed flat on the wall.  Hard to explain.  Right now it gets placed on wall in the orientation it was scanned, just stuck to the wall:
![Screen Shot 2023-12-01 at 11 58 13 am](https://github.com/google/model-viewer/assets/103499808/7596b119-88a6-4be4-a6b6-29186b8ad8c6)

Possible?  Or do I have to re-scan on a vertical surface?

**Comment by danilokrulj** [comment]
Why dont you just rotate the models accordingly a keep wall placement as it is? That should do the trick.

**Reply by sugarfreejaz** [reply]
is that with the code like orbit?  starting camera angles?

**Reply by danilokrulj** [reply]
No I mean rotate the model, open it in blender and rotate it in to the position you would like it to be.

**Reply by elalish** [reply]
You can also accomplish this with our `orientation` attribute: https://modelviewer.dev/docs/index.html#entrydocs-scenegraph-attributes-orientation

**Comment by Pruebas11111** [comment]
@sugarfreejaz How did you place the object on the ceiling? I can only do it on the floor or wall. Thank you!!

**Reply by sugarfreejaz** [reply]
ar="" ar-placement="wall"    maybe change wall to ceiling?     

my objects are placed on ceiling with the code provided.  The are placed there in the orientation in which they are scanned though.  since my objects are climbing holds I have been scanning the object to 3D on a vertical wall, thus when placed on any surface it is in that orientation.  so i guess if you want your objects placed on a ceiling in the correct orientation, you will have to scan the original object to 3D when it is on a ceiling.    this is an assumption ,i do not know.   you can scan the object in any orientation and then maybe use software like blender to change orientation?    

i know zero about code, maybe you can change the orientation of object depending if the surface is flat, vertical etc using model-viewer code?  beyond my knowledge.

**Reply by Pruebas11111** [reply]
@sugarfreejaz Hi. I had tried changing "wall" to "ceiling", and it doesn't work hehe. And the 3d model can be displayed "placed" on the ceiling (by moving the object with your fingers on the screen). But it still depends on whether you see the wall or the floor. And, when on the camera you no longer see the floor or the wall, it stops working. By the way, what I am trying to do is to place a lamp on the ceiling. but thanks anyway for the information and for your message!!!!!

**Reply by sugarfreejaz** [reply]
is the ceiling well lit?   i have noticed lighting needs to be very good and seems to me something on the surface to be placed has texture or other stuff nearby like light switch, art, door frame.

**Reply by Pruebas11111** [reply]
@sugarfreejaz No, the ceiling I used for testing did not have very good illumination. I'm going to try it with more light later. Thanks for the advice!
